src/fdosecrets/widgets/SettingsWidgetFdoSecrets.cpp

```diff
@@ -301,4 +312,53 @@ void SettingsWidgetFdoSecrets::saveSettings()
     FdoSecrets::settings()->setNoConfirmDeleteItem(m_ui->noConfirmDeleteItem->isChecked());
 }
 
+void SettingsWidgetFdoSecrets::showMessage(const QString& text, MessageWidget::MessageType type)
+{
+    // Show error messages for a longer time to make sure the user can read them
+    if (type == MessageWidget::Error) {
+        m_ui->warningMsg->setCloseButtonVisible(true);
+        m_ui->warningMsg->showMessage(text, type, -1);
+    } else {
+        m_ui->warningMsg->setCloseButtonVisible(false);
+        m_ui->warningMsg->showMessage(text, type, 2000);
+    }
+}
+
+void SettingsWidgetFdoSecrets::showEvent(QShowEvent* event)
+{
+    QWidget::showEvent(event);
+    QTimer::singleShot(0, this, &SettingsWidgetFdoSecrets::checkDBusName);
+    m_checkTimer.start();
+}
+
+void SettingsWidgetFdoSecrets::hideEvent(QHideEvent* event)
+{
+    QWidget::hideEvent(event);
+    m_checkTimer.stop();
+}
+
+void SettingsWidgetFdoSecrets::checkDBusName()
+{
+    if (m_plugin->serviceInstance()) {
+        // only need checking if the service is not started or failed to start.
+        return;
+    }
+
+    qDebug() << "Checking DBus Name";
```

Comment:
Do we need this?

Reply:
Oops, this was only needed for debugging.